app/support/schedule_rules/open_hours.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module ScheduleRules
+
+  class OpenHours
+    attr_reader :schedule_rules
+
+    def initialize(schedule_rules)
+      @schedule_rules = schedule_rules
+    end
+
+    def self.weekdays
+      Date::ABBR_DAYNAMES
+    end
+
+    def self.time_range(start_time, end_time)
+      [start_time, end_time].join(" - ")
+    end
+
+    def per_weekday
+      entries = rules_times.group_by { |entry| entry[:weekday] }
+
+      self.class.weekdays.index_with do |weekday|
+        entries[weekday]&.pluck(:range)&.join(", ")
+      end
+    end
+
+    def rules_times
+      return @rules_times if @rules_times.present?
+
+      @rules_times =
+        self.class.weekdays.flat_map do |weekday|
+          wday_rules = schedule_rules.filter { |sr| sr.on_day?(weekday) }
+
+          merged_rule_times(wday_rules).map do |time|
+            { weekday:, range: self.class.time_range(time[:start], time[:end]) }
+          end
+        end
+    end
+
+    private
+
+    def merged_rule_times(rules)
```

Comment:
Could you add a comment explaining what this is doing and what's the shape of the returned value?

Reply:
Hey @joaquinco you reacted with a 👍 but I think you forgot to make the changes 😄

Reply:
I think you need to check the latest changes